### DISCUSSION TITLE
How to differentiate between NextApiRequest and NextRequest?

### DISCUSSION THREAD

**Discussion by schrodienieur** [vercel/next.js#51614] [App Router]
```js
import { NextRequest } from 'next/server'
import type {NextApiRequest} from 'next'

export async function parseBody(req: NextRequest | NextApiRequest) {
  if (req instanceof Request) {
    console.log('NextRequest')
  } else {
    console.log('NextApiRequest')
  }
```

How to differentiate between NextRequest and NextApiRequest? Because the code above not working.

I use req instanceof Request, but both returning true.
I use req instanceof NextRequest, but it returning false when req: NextRequest.

**Comment by haithemyoucefkhoudja** [comment]
Hello, I have this solution:
we need  **res** parameter e.g.
`
destructor(req: NextApiRequest | NextRequest, res?:NextApiResponse)`

then we have to check if res has params property

`
function isAppRouter(res: any): boolean {
  return res.hasOwnProperty('params'); 
}
`
explanation:
in the app router version of next.js the API will pass this object to **res**

`  { params: undefined }`

while in page router version **res** will be NextApiResponse object.
